### ISSUE TITLE
sae_manifold_fit: no atom ever activates for K>=2 (constant R2 ~ 0, active=0)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#182]
## Observed
With `gamfit==0.1.123`, `sae_manifold_fit` does not learn anything when `K>=2`, regardless of `assignment_prior`, `sparsity_weight`, `n_iter`, or `learning_rate`. The returned `summary()` always reports `avg_active_atoms = 0.0` and a reconstruction R² that sits at the trivial 'predict the mean' baseline (negative, on the order of -0.002 to -0.32 depending on the dataset).

Only `K=1` actually fits the data.

## Sweep evidence (cluster data, 4 clusters in R^8, n=200)
```
mode=ibp_map    ni=20  sw=0.001..10  R2=-0.3182  active=0.00   (identical across all sw)
mode=jumprelu   ni=20  sw=0.001..10  R2=-0.3182  active=0.00
mode=gated      ni=20  sw=0.001..10  R2=-0.3182  active=0.00
mode=softmax    ...                 (separate crash, see #181)
```

## Sweep evidence (1-harmonic circle in R^32, n=400)
```
K=1 mode=ibp_map    R2=+0.21  active=1.00   <- only K=1 path activates
K=2 mode=ibp_map    R2=-0.00  active=0.00
K=4 mode=ibp_map    R2=-0.00  active=0.00
K=4 mode=jumprelu   R2=-0.00  active=0.00
K=4 mode=gated      R2=-0.00  active=0.00
```

The R² values being **bitwise identical across `sparsity_weight ∈ {1e-3, 0.1, 1, 10}` and across modes** strongly suggests no per-atom optimization is happening — atoms appear initialized off and never turned on.

## Expected
For a noise-free 1-harmonic circle embedded in R^32 with K=4 atoms of dimension 2, R² should approach 1 and at least one atom should be active (`avg_active_atoms >= ~1`). Different `sparsity_weight` values should change the active count and the score, not produce identical outputs.

## Minimal reproducer (≤ 30 lines)
```python
import numpy as np, gamfit
print("gamfit", gamfit.__version__)
rng = np.random.default_rng(0)
theta = rng.uniform(0, 2*np.pi, 400)
A = rng.standard_normal((2, 32)) * 0.5
X = np.stack([np.cos(theta), np.sin(theta)], axis=1) @ A + 0.05*rng.standard_normal((400,32))
for K in [1, 2, 4]:
    for mode in ["ibp_map", "jumprelu", "gated"]:
        m = gamfit.sae_manifold_fit(Z=X, K=K, d_atom=2, atom_topology="circle",
                                    assignment_prior=mode, n_iter=20,
                                    sparsity_weight=0.1, random_state=0)
        s = m.summary()
        print(f"K={K} mode={mode:10s} R2={s['reconstruction_r2']:+.4f}  active={s['avg_active_atoms']:.2f}")
```

Version: `gamfit 0.1.123` (pypi).

**Comment by SauersML** [comment]
Fixed by `cbedc218` (add LSQ decoder initialization in pyffi via `sae_decoder_lsq_init`) + `554daa4a` (K=2 manifold regression coverage + raw_inf type-inference fix).

Root cause: with the historical zero-initialized decoder, at iter 0 every atom's contribution to ∂L/∂A was proportional to `A · Φ` = 0, while the IBP / softmax assignment prior had a non-zero gradient pushing all rows toward a single atom. The first Newton step therefore drove the assignment to a one-hot point mass before the decoder could pick up any signal, and the joint fit stayed pinned at active=0 for K≥2 because the prior gradient at the saturated logit is also ~0.

Fix: the Python entry point now seeds the decoder blocks via a joint LSQ projection of Z onto `[a_init · Φ_k]` with a small ridge, so the very first decoder evaluation has non-zero residual gradient and the assignment prior no longer dominates the first step.

Regression coverage:
- `ibp_map_k2_periodic_torus_recovers_signal_with_lsq_init` in `src/terms/sae_manifold.rs:3034` — asserts both R² > 0.5 and `mean_active > 0.2` for K=2 IBP-MAP with LSQ-seeded decoder on a 2-circle torus signal.
- `softmax_k2_periodic_completes_joint_fit_step` in `src/terms/sae_manifold.rs:3210` — exercises the K=2 softmax companion path.

Closing as fixed.
